core/lowering/lowering.cpp

```diff
@@ -48,18 +48,26 @@ int AutocastLongInputs(
       auto dtype = dtype_input->second.value();
       // Currently, we do not autocast inputs for which the determined type is not long
       if (dtype != at::kLong) {
+        LOG_DEBUG(
+            "Skipping autocast for tensor " << input->debugName() << ", since its dtype is " << dtype
+                                            << " and not at::kLong");
         continue;
       }
 
       LOG_DEBUG("Inserting aten::to casting " << input->debugName() << " to dtype " << dtype);
 
       // Generate cast node sending input tensors to the inferred or specified datatype (long)
+      torch::jit::Value *const_false, *cuda, *none_val;
+      if (num_autocasts == 0) {
+        // Only generate constants once and reuse for all autocasts
+        const_false = g->insertConstant(0);
+        const_false->setType(torch::jit::BoolType::get());
+        cuda = g->insertConstant(target_device_name);
+        cuda->setType(torch::jit::DeviceObjType::get());
+        none_val = g->insertNode(g->createNone())->output();
+      }
+
       auto const_type = g->insertConstant(dtype);
```

Comment:
Should this be dtype instead of INT32 ? coz dtype here would be at::kLong correct ?

Reply:
It is correct that `dtype = at::kLong` in this case, but the intent was to cast to `at::kLong` here, because this autocast was written to insert a Torch block at the beginning of the graph.

The reason is for the following case: if partitioning determines that the first block executed in the graph should be TensorRT, but the user specifies an INT64 input type, we currently can't insert a cast to INT32 without adding a Torch-executed block first. By casting to INT64 and marking the cast node to run in PyTorch, we ensure that a Torch block is inserted first, and partitioning's INT64 <-> INT32 conversion takes care of the rest.